### ISSUE TITLE
report existing annotations which would not be covered by AE config file 

### ISSUE THREAD

**Issue by ValWood** [pombase/pombase-chado#490]
One way to find out if anything is missing would be to report any  AE's which do not fit the patterns allowed in the config. I know there are a few 

If it isn't easy this isn't high priority, because as we come across things we  are not prompted for  we can extend the configs

**Comment by kimrutherford** [comment]
It's a bit fiddly but it will need to be done.  At some point we will need to update all the old extensions to work in the in interface.  When that code is written it will need to check the config file to make sure the existing extensions make sense.

**Comment by kimrutherford** [comment]
> At some point we will need to update all the old extensions to work in the in interface. 

Actually that only helps for the extensions from Canto so we'll need something more general.

**Comment by ValWood** [comment]
Is this done?

**Comment by ValWood** [comment]
If we already sanity check that the annotations in Canto use "allowed" extensions,
and only the artemis ones are not checked, I think we can put this at low priority?

In fact, they look pretty sensible (apart from the negating ones, but not a hurry):
https://curation.pombase.org/dumps/latest_build/logs/log.2016-10-19-00-29-27.extension_relation_counts

(I wonder what the part_of cellular component ones are?)

  191 | happens_during          | biological_process
   296 | has_input               | biological_process
   292 | has_regulation_target   | biological_process
     1 | not_happens_during      | biological_process
    26 | occurs_at               | biological_process
    12 | occurs_in               | biological_process
   166 | coincident_with         | cellular_component
  1164 | exists_during           | cellular_component
    15 | not_exists_during       | cellular_component
     8 | part_of                 | cellular_component
    67 | activated_by            | molecular_function
   169 | happens_during          | molecular_function
  2056 | has_direct_input        | molecular_function
   166 | has_regulation_target   | molecular_function
    67 | inhibited_by            | molecular_function
     3 | not_happens_during      | molecular_function
   103 | occurs_at               | molecular_function
    10 | occurs_in               | molecular_function
    94 | part_of                 | molecular_function

**Comment by kimrutherford** [comment]
>  I think we can put this at low priority?

Yep.  We currently check for "allowed" extension relations but we don't check that relations are only used on the right terms.  We also don't check the cardinality constrains from the config file.

I assume GO will check the GAF file we send them somehow?

**Comment by mah11** [comment]
> I assume GO will check the GAF file we send them somehow?

Ha ha. Well, for some relaxed values of "somehow", yes; and to be fair, it should improve in the foreseeable tho not immediate future.

At present the "official" GO Jenkins checks don't do anything with extensions as far as I can tell. Tony at GOA runs additional checks that do pay attention to extensions, but those haven't been incorporated into the Jenkins set yet (cue tangential rant about GOA and GOC downloads therefore differing), partly because they were still a bit buggy when I last checked and partly due to the usual big-project inertia.

It's on my post-holiday to-do list to look over the latest report from GOA and see which of the anomalies I spotted have been fixed. That doesn't address the "inertia" part of the scenario, but we can't do everything ;)

**Comment by ValWood** [comment]
@kimrutherford  I'm not sure if this is still a "valid" ticket.
I think you have probably addressed most extension  issues in subsequent tickets.
This one isn't a priority, but you could look and  give me a review of current extension usage so that I can open a new ticket if necessary describing any necessary actions.

**Comment by kimrutherford** [comment]
> but you could look and give me a review of current extension usage

What sort of information are you interested in?

On a recent call we discussed the idea of checking all extensions in Chado to make sure that all annotations follow the Canto extension configuration: https://github.com/pombase/pombase-config/tree/master/canto/annotation_ex_config

I think this issue is the same idea.

**Comment by kimrutherford** [comment]
> On a recent call we discussed the idea of checking all extensions in Chado to make sure that all annotations follow the Canto extension configuration

I thought this sounded familiar.  We talked about it here: https://github.com/pombase/canto/issues/2627#issuecomment-1328485477

**Comment by ValWood** [comment]
yep!

**Comment by kimrutherford** [comment]
Note to self: I've started work on this on a branch:
https://github.com/pombase/pombase-chado-json/tree/490-check-extenstions-with-canto-conf

It's probably half done.  I think it would need another half day or so to finish.

(I'm making a note here because I just found a bunch of changes from months ago I had stashed away locally that I had absolutely no recollection of.  I thought I better make a branch and a note to myself to help when I forget again)
